### DISCUSSION TITLE
Vercel deploy issue: Error: `dataset` must be provided to perform queries

### DISCUSSION THREAD

**Discussion by sdoigmm** [vercel/next.js#48849] [Help]
### Summary

I've created a Vercel project and linked it to a forked website Bit Bucket repo.

I've installed the vercel npm package and run vercel locally, logging into Vercel and linking the local project to my Vercel project.

I added the following environment variables to my Vercel project:

- dataset = production
- SANITY_API_DATASET = production
- SANITY_STUDIO_API_DATASET = production
- NEXT_PUBLIC_SANITY_DATASET = production

If I make a small change to my local project and git push, the Vercel deployment process starts, but the deployment fails with an error:

`Error: `dataset` must be provided to perform queries.`

There is a local `.env` file:

```
# Created by Vercel CLI
VERCEL="1"
VERCEL_ENV="development"
VERCEL_URL=""
VERCEL_GIT_PROVIDER=""
VERCEL_GIT_PREVIOUS_SHA=""
VERCEL_GIT_REPO_SLUG=""
VERCEL_GIT_REPO_OWNER=""
VERCEL_GIT_REPO_ID=""
VERCEL_GIT_COMMIT_REF=""
VERCEL_GIT_COMMIT_SHA=""
VERCEL_GIT_COMMIT_MESSAGE=""
VERCEL_GIT_COMMIT_AUTHOR_LOGIN=""
VERCEL_GIT_COMMIT_AUTHOR_NAME=""
SANITY_API_WRITE_TOKEN="xxx"
NEXT_PUBLIC_SANITY_PROJECT_ID="xxx"
SANITY_API_DATASET="production"
SANITY_STUDIO_API_PROJECT_ID="xxx"
NEXT_PUBLIC_SANITY_DATASET="production"
SANITY_API_PROJECT_ID="xxx"
SANITY_STUDIO_API_DATASET="production"
```
Help appreciated.

### Additional information

```js
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103
    Binaries:
      Node: 16.15.1
      npm: 8.11.0
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.0.2
      eslint-config-next: 13.0.2
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by icyJoseph** [comment]
I think you need to set the `PREVIEW` env vars in the Vercel Dashboard.

- https://vercel.com/docs/concepts/projects/environment-variables#declare-an-environment-variable

**Reply by sdoigmm** [reply]
I added the following env vars to Development, Preview, and Production;

- dataset
- SANITY_API_DATASET
- NEXT_PUBLIC_SANITY_DATASET
- SANITY_STUDIO_API_DATASET

and the error is gone now.

**Reply by NathanBrews** [reply]
this also worked for me seeding a user not good

**Reply by valtterisa** [reply]
this is not cool error to have in Vercel. @rauchg please fix
